packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -966,10 +976,13 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 							);
 						}
 					}
-					await Promise.all(attachBlobsP);
+					await Promise.race([Promise.all(attachBlobsP), abortPromise]).catch(() => {});
 				}
 
 				for (const [id, entry] of this.pendingBlobs) {
+					if (signal?.aborted && !entry.attached) {
```

Comment:
i think we need to make it clear somehow that this will return partial state in the event of the signal being cancelled, specifically that the signals cancels the waiting for blobs to attach, and not the generation of the pending state. this needs to be done at all layers

Reply:
we also need to think about debuggability here. i think we should log errors for all unattached blobs and ideally those errors are able to be correlated to creates, for this we might need some sort of logging id we track across the blobs Lifecyle.

Reply:
Clarity meaning comments/documentation here? 

Yeah I'm also going to add error logging as well for detached blobs. I'll think of a way to relate the error with the create flow as well.

Reply:
also naming, signal isn't very descriptive.